sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -78,6 +78,27 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
     }
   }
 
+  private def toSQL(node: LogicalPlan, topNode: Boolean): String = {
```

Comment:
I saw the test cases in HiveCompatibilitySuite failed. 
As @dilipbiswal said, `Limit` is one of the issues. `Filter` is another one. Besides these two, I also saw more. Thus, I did not continue to fix all of them.